compiler/rustc_hir_typeck/src/writeback.rs

```diff
@@ -720,17 +720,18 @@ impl<'cx, 'tcx> WritebackCx<'cx, 'tcx> {
             fcx_typeck_results.offset_of_data().items_in_stable_order()
         {
             let hir_id = HirId { owner: common_hir_owner, local_id };
-            let container = self.resolve(container, &hir_id);
+            let container = self.resolve(container, &hir_id, true);
             self.typeck_results.offset_of_data_mut().insert(hir_id, (container, indices.clone()));
         }
     }
 
-    fn resolve<T>(&mut self, value: T, span: &dyn Locatable) -> T
+    fn resolve<T>(&mut self, value: T, span: &dyn Locatable, should_normalize: bool) -> T
```

Comment:
having a bool argument for this seems confusing.

Please either change this to a custom enum or split this into `(normalize_and_)resolve` and `resolve_delayed_obligations`

Reply:
i think `normalize_and_resolve` + `resolve_delayed_coroutine_obligations` is fine. a bit of code duplication, but I also dislike all the `, true` additions.